src/vm/database/clarity_db.rs

```diff
@@ -101,11 +101,36 @@ pub trait BurnStateDB {
     fn get_pox_prepare_length(&self) -> u32;
     fn get_pox_reward_cycle_length(&self) -> u32;
     fn get_pox_rejection_fraction(&self) -> u64;
+
+    /// Returns the header hash of the burnchain block at burnchain height `height`
+    /// using the sortition `sortition_id`.
+    ///
+    /// Returns None if `height` is negative or greater than the canonical burn
+    /// chain height.
+    ///
+    /// # Panics
+    /// Panics if there is an error with the underlying database connection or
+    /// query.
     fn get_burn_header_hash(
         &self,
         height: u32,
         sortition_id: &SortitionId,
     ) -> Option<BurnchainHeaderHash>;
+
+    /// Returns the header hash of the burnchain block at burnchain height `height`
+    /// using the "canonical sortition" (see SortitionDB::get_canonical_sortition_tip).
+    ///
+    /// Returns None if `height` is negative or greater than the canonical burn
+    /// chain height.
+    ///
+    /// # Panics
+    /// Panics if there is an error with the underlying database connection or
+    /// query.
+    fn get_burn_header_hash_using_canonical_sortition(
+        &self,
+        height: u32,
+    ) -> Option<BurnchainHeaderHash>;
```

Comment:
You should probably just drop this method altogether.  This method is going to need to take the `StacksBlockId` of the chain tip that the VM is currently working on, in addition to the burnchain block height.  It would look something like this:

```
// just an example; feel free to rename
fn get_burn_header_hash_from_tip(&self, stacks_tip: &StacksBlockId, burn_height: u32) -> Option<BurnchainHeaderHash>)
```